Dockerfile

```diff
@@ -41,14 +51,16 @@ ARG torch_cuda_arch_list='7.0 7.5 8.0 8.6 8.9 9.0+PTX'
 ENV TORCH_CUDA_ARCH_LIST=${torch_cuda_arch_list}
 #################### BASE BUILD IMAGE ####################
 
-
 #################### WHEEL BUILD IMAGE ####################
-FROM dev AS build
+FROM base AS build
+
+ARG PYTHON_VERSION=3
```

Comment:
Does multi-stage dockerifle support global arg?

Reply:
I don't think so. `ARG` defined before `FROM` command can be only used for the `FROM` command IIRC. For multi-stage, here is what said on Docker documentation:

> An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an argument in multiple stages, each stage must include the ARG instruction.

Reply:
Just as a fyi: it sort of does. What you have to do is define all your desired global `ARG`s at the top of the Dockerfile, before any `FROM`s. Then, for each stage which uses such a global arg, you'll also have to add in an empty `ARG` to import the value of that global arg into that stage, e.g. just a `ARG PYTHON_VERSION` here.